lmcache/cache_engine.py

```diff
@@ -301,17 +308,29 @@ def retrieve(
                 
                 For vllm, it should have the shape of 
                 [num_tokens, num_heads, head_size]
+                
+            mask: a boolean mask of tokens indicating which tokens'
+            KV Cache should be retrieved
 
         Output:
             kv_tensors: the kv cache of the tokens, in the format of nested 
                         tuples. Will be an empty tuple if no kv cache is 
                         retrieved.
 
-            num_tokens: the number of tokens in the kv cache
+            ret_mask: indicate which tokens are retrieved
         """
+        num_skip_chunk = 0
+        num_skip_tok = 0
+        ret_mask = torch.ones_like(tokens, dtype=torch.bool)
+        if mask is not None:
+            num_skip_tok = (len(mask) - torch.sum(mask))
+            num_skip_chunk = num_skip_tok // self.chunk_size
```

Comment:
What if the `mask` does not align with the chunk boundary? Will the `num_skip_chunk` still make sense?
Please clarify if you had any extra assumption on `mask`

Reply:
`num_skip_chunk` will still make sense. For example, if the mask is xxxvvv (only want to retrieve the last three tokens) and the chunk_size is 2, then we will skip the first chunk (the first two tokens) and still retrieve the third token even though it's x. This is because we save the cache chunk by chunk.

Reply:
I see, so it should work for any "suffix mask", right? Do you support (or plan to support) arbitrary masks (like xxvvxvxv)?

Reply:
Yes. But I think we should have a separate pr for this. Currently, this is only needed for (1) chunk prefill and (2) compatibility with vllm's internal prefix caching. Suffix mask would suffice for both cases.

Reply:
Sounds great! Can you maybe add a few sentences in the docstring about this? (I mean currently the mask should be "suffix")

Reply:
Done.